packages/transaction-controller/src/TransactionController.ts

```diff
@@ -2458,6 +2459,7 @@ export class TransactionController extends BaseController<
       {
         transactionId,
         note: 'TransactionController#updateSimulationData - Update simulation data',
+        skipHistory: this.isHistoryDisabled,
```

Comment:
Since mobile doesn't enable the `history` I believe this change also need to be included in the core transaction controller. 
https://github.com/MetaMask/core/pull/4349

Reply:
I've suggested an alternate fix in the `main` PR that could also apply here.

Reply:
Updated both, thanks for flagging this